syntax = "proto3";
package voxel.dialogue.v1;

import "voxel/common/v1/types.proto";

message AskRequest {
  string guru_id = 1;
  string user_query = 2;              // text input (or use audio field for ASR)
  bool   return_audio = 3;            // if true, stream TTS audio chunks
}

message AskChunk {
  // Server-streaming chunks as the LLM/TTS progresses
  oneof payload {
    string transcript_delta = 1;      // incremental text
    voxel.common.v1.LipSync lipsync = 2; // full or incremental marks
    bytes  audio_chunk = 3;           // optional TTS audio
  }
  bool is_final = 10;
}

service DialogueService {
  // Server streaming: send a question, receive transcript/phoneme/audio chunks
  rpc Ask(AskRequest) returns (stream AskChunk);

  // Optional bidi (for continuous back-and-forth or voice input):
  // rpc Converse(stream AskRequest) returns (stream AskChunk);
}
